### PR TITLE
[aotinductor] Relax ExternKernel kwargs checking

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #111167

Summary: When a fallback kernel is called without specifying any kwargs, we still need to fill in default values for those kwargs when generating cpp call.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: test/inductor/test_aot_inductor.py [hunk 1]
```diff
@@ -143,10 +143,13 @@ def check_model(
     with torch.no_grad(), config.patch(
         "aot_inductor.abi_compatible", self.abi_compatible
     ):
+        torch.manual_seed(0)
         model = model.to(self.device)
         ref_model = copy.deepcopy(model)
         ref_inputs = copy.deepcopy(example_inputs)
         expected = ref_model(*ref_inputs)
+
+        torch.manual_seed(0)
```
Comment (chenyang78) [inline test/inductor/test_aot_inductor.py:152]: Wondering why do we need the second call to ```manual_seed```?

**Review by chenyang78 — APPROVED** [review]

File: test/inductor/test_aot_inductor.py (hunk 1 above)
Comment (desertfire) [reply test/inductor/test_aot_inductor.py:152]: Because we need to compare the results of eager and compiled, resetting seed makes sure both will generate the same random numbers.

Comment (chenyang78) [reply test/inductor/test_aot_inductor.py:152]: Hmm, we actually generate example_inputs before invoking these two manual_seed, i.e. example inputs are passed into the ```check_model``` function. Is there any hidden state that required random numbers?

Comment (desertfire) [reply test/inductor/test_aot_inductor.py:152]: Yeah, in the new test I added.

**Review by jgong5 — APPROVED** [review]
